ros1_bridge/__init__.py

```diff
@@ -323,6 +402,79 @@ def determine_message_pairs(ros1_msgs, ros2_msgs, package_pairs, mapping_rules):
     return pairs
 
 
+def determine_common_services(ros1_srvs, ros2_srvs, mapping_rules):
+    pairs = []
+    services = []
+    for ros1_srv in ros1_srvs:
+        for ros2_srv in ros2_srvs:
+            if (ros1_srv.package_name == ros2_srv.package_name):
+                if (ros1_srv.message_name == ros2_srv.message_name):
+                    pairs.append((ros1_srv, ros2_srv))
+
+    for rule in mapping_rules:
+        for ros1_srv in ros1_srvs:
+            for ros2_srv in ros2_srvs:
+                if rule.ros1_package_name == ros1_srv.package_name and \
+                   rule.ros2_package_name == ros2_srv.package_name:
+                    if rule.ros1_service_name == None and rule.ros2_service_name == None:
+                        if (ros1_srv.message_name == ros2_srv.message_name):
+                            pairs.append((ros1_srv, ros2_srv))
+                    else:
+                        if rule.ros1_service_name == ros1_srv.message_name and \
+                           rule.ros2_service_name == ros2_srv.message_name:
+                            pairs.append((ros1_srv, ros2_srv))
+
+    for pair in pairs:
+        ros1_spec = load_ros1_service(pair[0])
+        ros2_spec = load_ros2_service(pair[1])
+        ros1_fields = {
+            "request": ros1_spec.request.fields(),
+            "response": ros1_spec.response.fields()
+        }
+        ros2_fields = {
+            "request": ros2_spec.request.fields,
+            "response": ros2_spec.response.fields
+        }
+        output = {
+            "request": [],
+            "response": []
+        }
+        match = True
+        for direction in ["request", "response"]:
+            if len(ros1_fields[direction]) != len(ros2_fields[direction]):
+                match = False
+                break
+            for i, ros1_field in enumerate(ros1_fields[direction]):
+                ros1_type = ros1_field[0]
+                ros2_type = str(ros2_fields[direction][i].type)
+                ros1_name = ros1_field[1]
+                ros2_name = ros2_fields[direction][i].name
+                if (ros1_type != ros2_type or ros1_name != ros2_name):
+                    match = False
+                    break
+                output[direction].append({
+                    "basic": False if "/" in ros1_type else True,
+                    "array": True if "[]" in ros1_type else False,
+                    "ros1": {
+                        "type": ros1_type.rstrip("[]"),
+                        "name": ros1_name,
+                    },
+                    "ros2": {
+                        "type": ros2_type.rstrip("[]"),
+                        "name": ros2_name,
+                    }
+                })
```

Comment:
As I mentioned before, it is possible to unify some functions in this script but it might not be that easy and I do not have time to do it now.